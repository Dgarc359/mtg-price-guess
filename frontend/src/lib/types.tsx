export type Card = {
  name: string,
  image_uris: {
    png: string,
    art_crop: string,
  },
  oracle_text: string,
  prices: {
    usd: string,
    usdFoil: string | null
  }
}