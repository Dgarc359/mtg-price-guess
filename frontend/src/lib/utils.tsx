import { GameChoice } from "../components/state-orchestrator";
import { CommonCard, MtgCard, PokemonCard } from "./types";


export const cn = (classNames: string[]) => {
  return classNames.join(" ").trim();
}

export const createCommonCard = (
  gameChoice: GameChoice,
  card: MtgCard | PokemonCard | undefined
): CommonCard | undefined => {
  if(!card) { 
    return card;
  }
  switch (gameChoice) {
    case "Magic The Gathering": {
      return card as CommonCard;
    }
    case "Pokemon": {
      const c = card as any[0] as PokemonCard;
      console.log('card',c);
      return {
        name: c.name,
        image_uris: {
          png: c.images.small,
          art_crop: c.images.large,
        },
        prices: {
          usd: c.cardmarket.prices.averageSellPrice.toString(),
          usdFoil: null,
        },
        legalities: {
          modern: c.legalities.unlimited as any
        },
        set_name: c.set.name
      };
    }
  }
};