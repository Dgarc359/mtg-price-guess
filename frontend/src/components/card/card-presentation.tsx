import { Card as MuiCard, CardMedia, CardContent, Typography, CardActionArea, Paper, Grid, IconButton } from "@mui/material";
import React from "react";
import { Card } from "../../lib/types";
import { cn } from "../../lib/utils";

interface ICard {
  cardData: Card | undefined;
  // imageOnClick?: (event?: MouseEvent<HTMLImageElement, MouseEvent>) => void,
  imageOnClick?: (event?: unknown) => void;
}

export const CardView = (props: ICard) => {
  const {
    imageOnClick,
  } = props;

  return (
    <div
      id={`card-container`}
      className={cn([
        "p-4 m-4"
        // "flex",
        // "max-sm:w-full max-sm:h-[50vh]",
        // "md:w-1/2 md:h-screen md:justify-center md:align-middle md:8",
      ])}
    >
      {/* <img
        src={props.cardData?.image_uris?.art_crop}
        alt=""
        className="blur-xl"
      /> */}

      {/* <div
        id="div-img-container"
        className={cn([
          "absolute",
          "max-sm:h-1/2", 
        ])}
      > */}
        {/* <img
          id={props.cardData?.name}
          src={props.cardData?.image_uris?.png}
          alt=""
          className={cn([
            "hover:cursor-pointer",
            "max-sm:w-2/6",
            "md:w-1/4 md:h-1/2 md:m-auto",
          ])}
          onClick={imageOnClick}
        /> */}
        {/* <div className={cn([
          "max-sm:absolute",
          "w-1/4 text-center m-auto mt-8"
        ])}>
          <div className={cn([
            // "bg-red-100",
            "bg-white",
            "before:absolute before:top-0 before:left-0 before:right-0 before:bottom-0",
            "shadow-2xl rounded-xl",
            // "blur-md opacity-75"
            "relative overflow-hidden"
          ])}>
            <div id="card-name" className="text-center text-3xl p-2">
              {props.cardData?.name}
            </div>
            <div id="card-action" className={cn([
              "md:text-center p-2"
            ])}>
              {props.cardData?.oracle_text}
            </div>
           
          </div>
         
        </div> */}
        {/* <MuiCard sx={{ 
            // width: 400, 
            width: {
              xs: 300,
              md: 400
            }
            // maxWidth: "50%",
            // minWidth: "50%",
        }}>
          <CardActionArea onClick={imageOnClick} id={props.cardData?.name}>
          <img
          id={props.cardData?.name}
          src={props.cardData?.image_uris?.art_crop}
          alt=""
          className={cn([
            "hover:cursor-pointer",
            "max-sm:w-2/6",
            "md:w-full md:h-full md:m-auto",
          ])}
          />
          <CardContent>
            <Typography gutterBottom variant="h5" component={"div"}>
              {props.cardData?.name}
            </Typography>
            <Typography variant="body2" color={"text.secondary"}>
              {props.cardData?.oracle_text}
            </Typography>
          </CardContent>
          </CardActionArea>
        </MuiCard> */}
      {/* </div> */}
      {/* <MuiCard sx={{width: {sm: 300, md: 400}}}>
      <CardActionArea>
      <CardContent> */}
      <IconButton>
      <Paper 
        sx={{
          p:2,
          margin: "auto",
          // maxWidth: 500,
          flexGrow: 1,
          backgroundColor: "#fff",
          width: {
            sm: 300,
            md: 400,
          },
          cursor: "pointer"
        }}
        onClick={imageOnClick}
        id={props.cardData?.name}
      >
        <Grid container spacing={2}>
          <Grid item md={12} xs={4}>
            <img
            id={props.cardData?.name}
            src={props.cardData?.image_uris?.png}
            alt=""
            className={cn([
              // "hover:cursor-pointer",
              "max-sm:w-[90%]",
              "md:w-[70%] md:m-auto",
            ])}
            />
         
          </Grid>
          <Grid item md={12} xs={8}>
            <Typography gutterBottom variant="h5" component={"div"} sx={{textAlign: {sm: "left", md: "center"}}}>
              {/* {props.cardData?.name} */}
              Quick Stats
            </Typography>
            <Typography variant="body2" color={"text.secondary"} sx={{overflow: "hidden", whiteSpace: "nowrap", textAlign: { sm: "left", md: "center"}}}>
              {`Card Name: ${props.cardData?.name}`}
            </Typography>
            <Typography variant="body2" color={"text.secondary"} sx={{textAlign: {sm: "left", md: "center"}}}>
              {/* {props.cardData?.oracle_text} */}
              {props.cardData?.legalities.modern === "legal" ? "Legal in Modern" : "Illegal in Modern"}
            </Typography>
            
            <Typography variant="body2" color={"text.secondary"} sx={{textAlign: {sm: "left", md: "center"}}}>
              {`Set: ${props.cardData?.set_name}`}
            </Typography>
          </Grid>

        </Grid>
      </Paper>
      </IconButton>
      {/* </CardContent>
      </CardActionArea> 
      </MuiCard> */}
    </div>
  );
};
