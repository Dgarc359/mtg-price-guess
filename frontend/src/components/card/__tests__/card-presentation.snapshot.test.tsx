// import type * as Render from 'react-test-renderer'
// const {TestRenderer} = jest.requireActual<typeof Render>('react-test-renderer');
// import TestRenderer from 'react-test-renderer';
import * as renderer from 'react-test-renderer';
import {CardView} from '../card-presentation';
import { Card } from '../../../lib/types';
import {test, expect} from 'vitest';

    test("tests snapshot", () => {
        const cardData: Card = {
            name: "test card",
            image_uris: {
                png: "png url",
                art_crop: "art_crop url"
            },
            oracle_text: "fake oracle text",
            prices: {
                usd: "100",
                usdFoil: "100",
            },
            legalities: {
                modern: "legal",
            },
            set_name: "fake set name"
        }
        // const card = TestRenderoer
        const card = renderer
            .create(<CardView cardData={cardData}/>)
            .toJSON();
        expect(card).toMatchSnapshot();
    })
