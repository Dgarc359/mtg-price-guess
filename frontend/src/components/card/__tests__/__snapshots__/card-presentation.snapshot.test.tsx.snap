// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests snapshot 1`] = `
<div
  className="p-4 m-4"
  id="card-container"
>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-da3c9u-MuiIconButton-root{text-align:center;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;font-size:1.5rem;padding:8px;border-radius:50%;overflow:visible;color:rgba(0, 0, 0, 0.54);-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-da3c9u-MuiIconButton-root:hover{background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-da3c9u-MuiIconButton-root:hover{background-color:transparent;}}.css-da3c9u-MuiIconButton-root.Mui-disabled{background-color:transparent;color:rgba(0, 0, 0, 0.26);}",
      }
    }
    data-emotion="css da3c9u-MuiIconButton-root"
  />
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-78trlr-MuiButtonBase-root-MuiIconButton-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;text-align:center;-webkit-flex:0 0 auto;-ms-flex:0 0 auto;flex:0 0 auto;font-size:1.5rem;padding:8px;border-radius:50%;overflow:visible;color:rgba(0, 0, 0, 0.54);-webkit-transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;}.css-78trlr-MuiButtonBase-root-MuiIconButton-root::-moz-focus-inner{border-style:none;}.css-78trlr-MuiButtonBase-root-MuiIconButton-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-78trlr-MuiButtonBase-root-MuiIconButton-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-78trlr-MuiButtonBase-root-MuiIconButton-root:hover{background-color:rgba(0, 0, 0, 0.04);}@media (hover: none){.css-78trlr-MuiButtonBase-root-MuiIconButton-root:hover{background-color:transparent;}}.css-78trlr-MuiButtonBase-root-MuiIconButton-root.Mui-disabled{background-color:transparent;color:rgba(0, 0, 0, 0.26);}",
      }
    }
    data-emotion="css 78trlr-MuiButtonBase-root-MuiIconButton-root"
  />
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1xs3oga-MuiPaper-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border-radius:4px;box-shadow:0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);padding:16px;margin:auto;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;background-color:#fff;cursor:pointer;}@media (min-width:600px){.css-1xs3oga-MuiPaper-root{width:300px;}}@media (min-width:900px){.css-1xs3oga-MuiPaper-root{width:400px;}}",
        }
      }
      data-emotion="css 1xs3oga-MuiPaper-root"
    />
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1xs3oga-MuiPaper-root"
      id="test card"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-mhc70k-MuiGrid-root{box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex-wrap:wrap;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;width:100%;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;margin-top:-16px;width:calc(100% + 16px);margin-left:-16px;}.css-mhc70k-MuiGrid-root>.MuiGrid-item{padding-top:16px;}.css-mhc70k-MuiGrid-root>.MuiGrid-item{padding-left:16px;}",
          }
        }
        data-emotion="css mhc70k-MuiGrid-root"
      />
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1vr1qqn-MuiGrid-root{box-sizing:border-box;margin:0;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-basis:33.333333%;-ms-flex-preferred-size:33.333333%;flex-basis:33.333333%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:33.333333%;}@media (min-width:600px){.css-1vr1qqn-MuiGrid-root{-webkit-flex-basis:33.333333%;-ms-flex-preferred-size:33.333333%;flex-basis:33.333333%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:33.333333%;}}@media (min-width:900px){.css-1vr1qqn-MuiGrid-root{-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:100%;}}@media (min-width:1200px){.css-1vr1qqn-MuiGrid-root{-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:100%;}}@media (min-width:1536px){.css-1vr1qqn-MuiGrid-root{-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:100%;}}",
            }
          }
          data-emotion="css 1vr1qqn-MuiGrid-root"
        />
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiGrid-grid-md-12 css-1vr1qqn-MuiGrid-root"
        >
          <img
            alt=""
            className="max-sm:w-[90%] md:w-[70%] md:m-auto"
            id="test card"
            src="png url"
          />
        </div>
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-epupbn-MuiGrid-root{box-sizing:border-box;margin:0;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-basis:66.666667%;-ms-flex-preferred-size:66.666667%;flex-basis:66.666667%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:66.666667%;}@media (min-width:600px){.css-epupbn-MuiGrid-root{-webkit-flex-basis:66.666667%;-ms-flex-preferred-size:66.666667%;flex-basis:66.666667%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:66.666667%;}}@media (min-width:900px){.css-epupbn-MuiGrid-root{-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:100%;}}@media (min-width:1200px){.css-epupbn-MuiGrid-root{-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:100%;}}@media (min-width:1536px){.css-epupbn-MuiGrid-root{-webkit-flex-basis:100%;-ms-flex-preferred-size:100%;flex-basis:100%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:100%;}}",
            }
          }
          data-emotion="css epupbn-MuiGrid-root"
        />
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-md-12 css-epupbn-MuiGrid-root"
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1o4v5k-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:1.5rem;line-height:1.334;letter-spacing:0em;margin-bottom:0.35em;}@media (min-width:600px){.css-1o4v5k-MuiTypography-root{text-align:left;}}@media (min-width:900px){.css-1o4v5k-MuiTypography-root{text-align:center;}}",
              }
            }
            data-emotion="css 1o4v5k-MuiTypography-root"
          />
          <div
            className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1o4v5k-MuiTypography-root"
          >
            Quick Stats
          </div>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-264ae3-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;color:rgba(0, 0, 0, 0.6);overflow:hidden;white-space:nowrap;}@media (min-width:600px){.css-264ae3-MuiTypography-root{text-align:left;}}@media (min-width:900px){.css-264ae3-MuiTypography-root{text-align:center;}}",
              }
            }
            data-emotion="css 264ae3-MuiTypography-root"
          />
          <p
            className="MuiTypography-root MuiTypography-body2 css-264ae3-MuiTypography-root"
          >
            Card Name: test card
          </p>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1dsvagk-MuiTypography-root{margin:0;font-family:\\"Roboto\\",\\"Helvetica\\",\\"Arial\\",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;color:rgba(0, 0, 0, 0.6);}@media (min-width:600px){.css-1dsvagk-MuiTypography-root{text-align:left;}}@media (min-width:900px){.css-1dsvagk-MuiTypography-root{text-align:center;}}",
              }
            }
            data-emotion="css 1dsvagk-MuiTypography-root"
          />
          <p
            className="MuiTypography-root MuiTypography-body2 css-1dsvagk-MuiTypography-root"
          >
            Legal in Modern
          </p>
          <p
            className="MuiTypography-root MuiTypography-body2 css-1dsvagk-MuiTypography-root"
          >
            Set: fake set name
          </p>
        </div>
      </div>
    </div>
  </button>
</div>
`;
